### ISSUE TITLE
Parser error for 'or' operator in if/while let expressions

### ISSUE THREAD

**Issue by iffyio** [intellij-rust/intellij-rust#4843]
## Environment

* **IntelliJ Rust plugin version:** 0.2.0.2114-182
* **Rust toolchain version:** rustc 1.39.0 (4560ea788 2019-11-04)
* **IDE name and version:** Intellij 2018.2
* **Operating system:** Linux Mint 19 Tara

## Problem description
Using the `or` operator during pattern matching in an `if let` or `while let` expression
is flagged as a syntax error within the IDE even though it is valid code.

## Steps to reproduce
e.g the following code compiles but is flagged as an error by the plugin:
```rust
fn main() {
  if let 1 | 2 = 3 {}
}
```

**Comment by Undin** [comment]
@iffyio it was fixed in #3447.
But you are using too old plugin version released year ago that doesn't contain this fix.
Please, update the plugin.
Note, the latest plugin versions are available only for 2019.2 and 2019.3 IDEs
